src/server.cpp

```diff
@@ -1572,6 +1538,76 @@ void CServer::WriteHTMLChannelList()
     streamFileOut << "</ul>" << endl;
 }
 
+void CServer::RequestNewRecording()
+{
+    if ( bRecorderInitialised && bEnableRecording )
+    {
+        emit RestartRecorder();
+    }
+}
+
+void CServer::SetEnableRecording ( bool bNewEnableRecording )
+{
+    if ( bRecorderInitialised ) {
+        bEnableRecording = bNewEnableRecording;
+        if ( !bEnableRecording )
+        {
+            emit StopRecorder();
+        }
+        else if ( !IsRunning() )
+        {
+            // This dirty hack is for the GUI.  It doesn't care.
+            emit StopRecorder();
+        }
+    }
+}
+
+void CServer::SetRecordingDir ( QString newRecordingDir )
```

Comment:
OK, that'll need another class as for thread management with an instance that lives in the server thread.

I have a very bad feeling that previously I was taking the whole server with me when I did "moveToThread"

Reply:
One more comment: My wish would be that if the recorder is not in use, it will not use any resources in the server. I.e there are no new threads, memory allocations, etc.

Reply:
So I don't forget, I'll say it again clearly:
* New JamManager class
* That gets all the code (and supporting variables) from:
```
    void RestartRecorder();
    void StopRecorder();
    void RecordingSessionStarted ( QString sessionDir );
    void EndRecorderThread();
    bool GetRecorderInitialised() { return bRecorderInitialised; }
    QString GetRecorderErrMsg() { return strRecorderErrMsg; }
    bool GetRecordingEnabled() { return bEnableRecording; }
    void SetEnableRecording ( bool bNewEnableRecording );
    QString GetRecordingDir() { return strRecordingDir; }
    void SetRecordingDir( QString newRecordingDir );
```
* Expose as necessary with simple pass through methods
* Expose any signals with simple pass through signals
* Looks like my "dirty hack" to get recorder stop to the GUI will need cleaning; there's a server Stopped() signal, which I should probably have used, I think.

Reply:
Ok, sounds good :-).